packages/pds/src/sql-repo-storage.ts

```diff
@@ -143,205 +150,108 @@ export class SqlRepoStorage extends RepoStorage {
-      for (const block of commit.blocks.entries()) {
-        commitCids.push(block.cid)
-        allBlocks.set(block.cid, block.bytes)
-      }
-      cidData.push({
-        commit: commit.commit,
-        prev: commit.prev,
-        cids: commitCids,
-      })
-    }
-    await Promise.all([this.putMany(allBlocks), this.indexCommitCids(cidData)])
-  }
-
-  async indexCommitCids(commits: CommitCidData[]): Promise<void> {
-    this.db.assertTransaction()
-    const commitBlocks: RepoCommitBlock[] = []
-    const commitHistory: RepoCommitHistory[] = []
-    for (const commit of commits) {
-      for (const cid of commit.cids) {
-        commitBlocks.push({
-          commit: commit.commit.toString(),
-          block: cid.toString(),
-          creator: this.did,
-        })
-      }
-      commitHistory.push({
-        commit: commit.commit.toString(),
-        prev: commit.prev ? commit.prev.toString() : null,
-        creator: this.did,
-      })
-    }
-    const insertCommitBlocks = Promise.all(
-      chunkArray(commitBlocks, 500).map((batch) =>
-        this.db.db
-          .insertInto('repo_commit_block')
-          .values(batch)
-          .onConflict((oc) => oc.doNothing())
-          .execute(),
-      ),
-    )
-    const insertCommitHistory = Promise.all(
-      chunkArray(commitHistory, 500).map((batch) =>
-        this.db.db
-          .insertInto('repo_commit_history')
-          .values(batch)
-          .onConflict((oc) => oc.doNothing())
-          .execute(),
-      ),
-    )
-    await Promise.all([insertCommitBlocks, insertCommitHistory])
+  async applyCommit(commit: CommitData) {
+    await Promise.all([
+      this.updateRoot(commit.cid, commit.prev ?? undefined),
+      this.putMany(commit.newBlocks, commit.rev),
+      this.deleteMany(commit.removedCids.toList()),
+    ])
   }
 
-  async updateHead(cid: CID, prev: CID | null): Promise<void> {
-    if (prev === null) {
-      await this.db.db
-        .insertInto('repo_root')
-        .values({
-          did: this.did,
-          root: cid.toString(),
-          indexedAt: this.getTimestamp(),
-        })
-        .execute()
-    } else {
+  async updateRoot(cid: CID, ensureSwap?: CID): Promise<void> {
+    if (ensureSwap) {
       const res = await this.db.db
         .updateTable('repo_root')
         .set({
           root: cid.toString(),
           indexedAt: this.getTimestamp(),
         })
         .where('did', '=', this.did)
-        .where('root', '=', prev.toString())
+        .where('root', '=', ensureSwap.toString())
         .executeTakeFirst()
       if (res.numUpdatedRows < 1) {
         throw new ConcurrentWriteError()
       }
+    } else {
+      await this.db.db
+        .insertInto('repo_root')
+        .values({
+          did: this.did,
+          root: cid.toString(),
+          indexedAt: this.getTimestamp(),
+        })
+        .onConflict((oc) =>
+          oc.column('did').doUpdateSet({ root: cid.toString() }),
```

Comment:
Small thing, but do we want to update `indexedAt` too?

Reply:
ya good call 👍